sound/soc/sof/ipc4-pcm.c

```diff
@@ -62,10 +62,36 @@ int sof_ipc4_set_pipeline_state(struct snd_sof_dev *sdev, u32 instance_id, u32 s
 }
 EXPORT_SYMBOL(sof_ipc4_set_pipeline_state);
 
+static void sof_ipc4_add_pipeline_by_priority(struct ipc4_pipeline_set_state_data *trigger_list,
+					      struct snd_sof_widget *pipe_widget,
+					      s8 *pipe_priority, bool ascend)
+{
+	struct sof_ipc4_pipeline *pipeline = pipe_widget->private;
+	int i, j;
+
+	for (i = 0; i < trigger_list->count; i++) {
+		/* add pipeline from low priority to high */
+		if (ascend && pipeline->priority < pipe_priority[i])
+			break;
+		else if (!ascend && pipeline->priority > pipe_priority[i])
```

Comment:
again how do we deal with equal priorities, if they ever appear in a configuration?

Reply:
@plbossart we reserve the order for current logic since for equal priority pipelines, they will be added at the end of trigger array and don't insert in trigger array. Please check that " <" is used not "<=". Current CI test confirm that we don't have any regression issue.

Current logic is :
````c
	if (state == SOF_IPC4_PIPE_RUNNING || state == SOF_IPC4_PIPE_RESET)
		for (i = pipeline_list->count - 1; i >= 0; i--) {
			spipe = pipeline_list->pipelines[i];
			sof_ipc4_add_pipeline_to_trigger_list(sdev, state, spipe, trigger_list,
							      pipe_priority);
		}
	else
		for (i = 0; i < pipeline_list->count; i++) {
			spipe = pipeline_list->pipelines[i];
			sof_ipc4_add_pipeline_to_trigger_list(sdev, state, spipe, trigger_list,
							      pipe_priority);
		}
````

Reply:
It's very confusing, you have a comment "/* add pipeline from low priority to high */" but the result is exactly reversed. I can't follow what the order actually means.

You should really start by the *firmware* requirements.

Reply:
Add more comments. The comment is for if, not else.
````
		/* add pipeline from low priority to high */
		if (ascend && pipeline->priority < pipe_priority[i])
			break;
		/* add pipeline from high priority to low */
		else if (!ascend && pipeline->priority > pipe_priority[i])
			break;
````

Reply:
START and RESET add pipeline from high priority to low,  PAUSE adds pipeline from low priority to high